src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -213,18 +214,46 @@ class BoundedExecutor {
   boost::asio::thread_pool pool_;
 };
 
+/// Used by async actor mode. The fiber event will be used
+/// from python to switch control among different coroutines.
+/// Taken from boost::fiber examples
+/// https://github.com/boostorg/fiber/blob/7be4f860e733a92d2fa80a848dd110df009a20e1/examples/wait_stuff.cpp#L115-L142
+class FiberEvent {
+ public:
+  // Block the fiber until the event is notified.
+  void Wait() {
+    std::unique_lock<boost::fibers::mutex> lock(mutex);
+    cond.wait(lock, [this]() { return ready; });
+  }
+
+  // Notify the event and unblock all waiters.
+  void Notify() {
+    {
+      std::unique_lock<boost::fibers::mutex> lock(mutex);
+      ready = true;
+    }  // release mutex
+    cond.notify_one();
+  }
+
+ private:
+  boost::fibers::condition_variable cond;
+  boost::fibers::mutex mutex;
+  bool ready = false;
+};
+
 /// Used to ensure serial order of task execution per actor handle.
 /// See direct_actor.proto for a description of the ordering protocol.
 class SchedulingQueue {
  public:
   SchedulingQueue(boost::asio::io_service &main_io_service, DependencyWaiter &waiter,
-                  std::shared_ptr<BoundedExecutor> pool = nullptr,
+                  std::shared_ptr<BoundedExecutor> pool = nullptr, bool use_async = false,
```

Comment:
Instead of use_async, please rename them to use_asyncio here and elsewhere. This avoids confusion between normal async and using an async event loop. Asyncio isn't the perfect name but it is probably clear enough to readers.